provider/datagen/src/transform/cldr/units/mod.rs

```diff
@@ -49,53 +49,55 @@ impl DataProvider<UnitsConstantsV1Marker> for crate::DatagenProvider {
         // If CLDR added more constants that are defined in terms of other constants, the maximum depth should be increased.
         let maximum_depth = 10;
         let mut has_internal_constants;
-        for _ in 0..maximum_depth {
+        let mut max_depth_reached = 0;
+        while max_depth_reached < maximum_depth {
             has_internal_constants = false;
+            max_depth_reached += 1;
             let mut constants_with_constants_map_replaceable =
                 BTreeMap::<&str, (Vec<String>, Vec<String>, ConstantExactness)>::new();
             for (cons_name, (num, den, constant_exactness)) in constants_map_in_str_form.iter() {
-                let mut temp_num = num.clone();
-                let mut temp_den = den.clone();
+                let mut temp_num = Vec::<String>::new();
+                let mut temp_den = Vec::<String>::new();
                 let mut temp_constant_exactness = *constant_exactness;
 
-                for i in 0..temp_num.len() {
-                    if !contains_alphabetic_chars(temp_num[i].as_str())
-                        || is_scientific_number(temp_num[i].as_str())
-                    {
+                for num_str in num {
+                    if !contains_alphabetic_chars(num_str) || is_scientific_number(num_str) {
+                        temp_num.push(num_str.clone());
                         continue;
                     }
                     has_internal_constants = true;
 
                     if let Some((rnum, rden, rconstant_exactness)) =
-                        constants_map_in_str_form.get(temp_num[i].as_str())
+                        constants_map_in_str_form.get(num_str.as_str())
                     {
-                        temp_num.remove(i);
                         temp_num.extend(rnum.clone());
                         temp_den.extend(rden.clone());
-                        if *rconstant_exactness == ConstantExactness::Approximate {
+                        if rconstant_exactness == &ConstantExactness::Approximate {
                             temp_constant_exactness = ConstantExactness::Approximate;
                         }
+                    } else {
+                        temp_num.push(num_str.clone());
```

Comment:
isn't this an error case? you've already determined that it's not a number, so if the variable lookup fails you shouldn't try again on the next iteration.

Reply:
No, if the depth is two, it won't be there the first time. However, it will be there the next time.